internal/procedures/server/start.go

```diff
@@ -14,9 +14,13 @@ import (
 	"github.com/edulinq/autograder/internal/util"
 )
 
-func Start(initiator common.ServerInitiator) (err error) {
-	defer server.StopServer()
+type ProcedureServer struct {
```

Comment:
Because your start server function is not-exported,
I don't see too much use of this struct.
Unlike the API server, the functionality for working with the server in his package are functions (which are not all exported) and not methods.

I think the pattern here generally makes sense (and it makes sense that api/server and procedures/server have different patterns),
just get rid of this struct and store the api server instance instead.